applets/notification_area/na-grid.c

```diff
@@ -128,52 +236,32 @@ item_removed_cb (NaHost *host,
 
   gtk_container_remove (GTK_CONTAINER (self), GTK_WIDGET (item));
   self->items = g_slist_remove (self->items, item);
+  refresh_grid (self);
 }
 
 static void
-update_size_and_orientation (NaGrid         *self,
-                             GtkOrientation  orientation)
+na_grid_size_allocate (GtkWidget     *grid,
+                       GtkAllocation *allocation,
+                       gpointer       data)
 {
-  /* FIXME: do we really need that?  comes from NaTray */
-  /* FIXME: if we do, do that in overridden preferred size handlers */
-
-  /* note, you want this larger if the frame has non-NONE relief by default. */
-  switch (orientation)
-    {
-    case GTK_ORIENTATION_VERTICAL:
-      /* Give grid a min size so the frame doesn't look dumb */
-      gtk_widget_set_size_request (GTK_WIDGET (self), MIN_GRID_SIZE, -1);
-      break;
-    case GTK_ORIENTATION_HORIZONTAL:
-      gtk_widget_set_size_request (GTK_WIDGET (self), -1, MIN_GRID_SIZE);
-      break;
-    }
-}
-
-static void
-orientation_notify (GObject    *object,
-                    GParamSpec *pspec,
-                    gpointer    data)
-{
-  update_size_and_orientation (NA_GRID (object),
-                               gtk_orientable_get_orientation (GTK_ORIENTABLE (object)));
+  refresh_grid (NA_GRID (grid));
 }
 
 static void
 na_grid_init (NaGrid *self)
 {
-  GtkOrientation orientation;
-
   self->icon_padding = 0;
   self->icon_size = 0;
 
   self->hosts = NULL;
   self->items = NULL;
+  self->grid_props = g_new0 (GridProps, 1);
```

Comment:
You should free this in finalize.

Reply:
I just set the cols/rows properties on the `_NaGrid struct` directly, and removed the unused length property.